src/main/java/com/google/devtools/build/lib/runtime/commands/InfoCommand.java

```diff
@@ -216,6 +218,7 @@ public BlazeCommandResult exec(
 
       env.getEventBus().post(new NoBuildEvent());
       if (residue.size() > 0) {
+        ImmutableSet.Builder<String> unknownKeysBuilder = ImmutableSet.builder();
```

Comment:
This will crash if I specify the same unknown key twice in a row.

Reply:
Nope, it doesn't, see the new test below. Note that this is an ImmutableSet not an ImmutableMap.